pkg/watch/until.go

```diff
@@ -40,7 +41,7 @@ func Until(timeout time.Duration, watcher Interface, conditions ...ConditionFunc
 		after = time.After(timeout)
 	} else {
 		ch := make(chan time.Time)
-		close(ch)
+		defer close(ch)
```

Comment:
Am I reading this correctly? Without the `defer`, the `ConditionSucceeded` for loop would immediately return with `wait.ErrWaitTimeout`?